src/input.rs

```diff
@@ -17,23 +18,39 @@ impl<'a> Read for Input<'a> {
 
 impl<'a> Seek for Input<'a> {
     fn seek(&mut self, pos: SeekFrom) -> io::Result<u64> {
+        fn try_skip<R>(reader: R, pos: SeekFrom, err_desc: &'static str) -> io::Result<u64>
```

Comment:
Why is this implemented as a nested function instead of a separate function in the `impl` block? If you want to keep it as a inline function, maybe capture `pos` instead of passing it in twice?

Reply:
I would have made this a closure, except closures don't allow generics. Boo. I kept it as a nested function to keep the scope conservative -- but there's no issue with moving it to the `impl` scope, if you want.

Reply:
Ah, I see - thanks. Let's keep it as is.